sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsGdsStore.scala

```diff
@@ -100,11 +100,17 @@ class RapidsGdsStore(
       dst match {
         case dmOriginal: DeviceMemoryBuffer =>
           val dm = dmOriginal.slice(dstOffset, length)
+          val alignedSize = if (RapidsGdsStore.isAligned(srcOffset)) {
```

Comment:
Is this fallback necessary in practice?  Curious about the scenario if it is needed, because the entire premise of assuming it's safe to read past the end of the buffer is predicated on the aligned allocator being used, and checking the start address for 4K alignment does not guarantee the aligned allocator is being used.

Reply:
Turned this off for now.